extensions/interactions/TextInput/TextInput.js

```diff
@@ -81,3 +82,56 @@ oppia.directive('oppiaShortResponseTextInput', [
     };
   }
 ]);
+
+oppia.factory('textInputRulesService', ['$filter', function($filter) {
+  return {
+    Equals: function(answer, inputs) {
+      var normalizedAnswer = $filter('normalizeWhitespace')(answer);
+      var normalizedInput = $filter('normalizeWhitespace')(inputs.x);
+      return normalizedAnswer.toLowerCase() == normalizedInput.toLowerCase();
+    },
+    FuzzyEquals: function(answer, inputs) {
+      var normalizedAnswer = $filter('normalizeWhitespace')(answer);
+      var answerString = normalizedAnswer.toLowerCase();
+
+      var normalizedInput = $filter('normalizeWhitespace')(inputs.x);
+      var inputString = normalizedInput.toLowerCase();
+
+      if(inputString == answerString) return true;
+      var d = [];
+      for(var i = 0; i <= inputString.length; i++){
+        d.push([i]);
+      }
+      for(var j = 0; j <= answerString.length; j++){
+        d[0].push(j);
+      }
+      for(i = 1; i <= inputString.length; i++){
```

Comment:
Sorry my bad, I mixed up the input and the answer string when I tested that input. Theres a bug with my base case (0 gets pushed twice to d[0]), but I believe the recurrence is correct (d[i][j] holds the edit distance of inputString(0:i) and answerString(0:t)) : https://en.wikipedia.org/wiki/Wagner–Fischer_algorithm.

Hmm sorry what should the correct edit distance of 'hello' and 'abcde' be?